andela/travel_tool_front
# #162688568 Fix redirection to and from the error page

#### What does this PR do?
- This PR fixes the popping of the error page whenever a user submits wrong data when editing a particulare guesthouse's details.


#### Description of Task to be completed?
- Only dispatch the error page when the user tries to access an unexisting URL.also, only notify the user through a toast if they have submitted wrong data during editing of an existing guesthouse(submitting rooms with the same room names in the same guesthouse / editing a guesthouse with a name that already exists).
- Make the editing form resubmitable after the user has received the error toast.So that they can change the data and resubmit again.

#### How should this be manually tested?
- Clone the repo
    `git clone https://github.com/andela/travel_tool_front.git`
 - Check in to the project directory
   `cd travel_tool_front`
 - Checkout the **bg-fix-error-page-bug-#162688568** branch
   `bg-fix-error-page-bug-#162688568`
- login into the app as a super admin
- Navigate to the manage guest houses section and edit a guest house to have an already existing name or two rooms in it that have the same room.
- There should be an error message displayed on a toaster but the user should be able to edit to achieve the necessary information and resubmit the form

#### Any background context you want to provide?
- The error page bug was being caused by the presence of an error page that was being prompted by the existence of an error that came from the reducers i.e if a user has input the wrong form data.The presence of such an error page wasn't justified in the first place as such pages are should be rendered only if there's a wrong URL error.Furthermore, errors from reducers (in the accomodation page) have already been catered for through displaying them to the user as a toast.


#### What are the relevant pivotal tracker stories?

[#162688568](https://www.pivotaltracker.com/story/show/162688568)

#### Screenshots (if appropriate)
before: 

<img width="1045" alt="screen shot 2018-12-20 at 14 51 26" src="https://user-images.githubusercontent.com/19623571/50283387-d2b89c00-0466-11e9-91cb-7b81445a1ae0.png">

now:

<img width="1059" alt="screen shot 2018-12-20 at 14 50 25" src="https://user-images.githubusercontent.com/19623571/50283403-db10d700-0466-11e9-89c7-a52025399269.png">

#### Questions:
